src/neptune_scale/sync/sync_process.py

```diff
@@ -464,7 +343,32 @@ def _raise_exception(status_code: int) -> None:
         raise NeptuneUnexpectedResponseError()
 
 
-class StatusTrackingThread(Daemon, WithResources):
+def _merge_operations(operations: list[Operation]) -> tuple[UpdateRunSnapshots, SequenceId, datetime.datetime]:
+    snapshot_batch = UpdateRunSnapshots(snapshots=[op.operation for op in operations])  # type: ignore
+    return snapshot_batch, operations[-1].sequence_id, operations[-1].ts
+
+
+def _stream_operations(
+    operations_repository: OperationsRepository,
+    run_id: str,
+    project: str,
+    max_batch_size: int = 15 * 1024 * 1024,
+) -> Generator[tuple[RunOperation, SequenceId, datetime.datetime], None, None]:
+    while operations := operations_repository.get_operations(up_to_bytes=max_batch_size):
+        if operations[0].operation_type == OperationType.CREATE_RUN:
+            create_run = operations.pop(0)
+            operation = RunOperation(project=project, run_id=run_id, create=create_run.operation)  # type: ignore
+            yield operation, create_run.sequence_id, create_run.ts
+
+            if not operations:
+                continue
```

Comment:
We need some backoff here - 0.1s sleep for now.

Reply:
We use Daemon thread: https://github.com/neptune-ai/neptune-client-scale/blob/1c4cee5b435df00a7e4e58c97f270656ef706eff/src/neptune_scale/util/daemon.py#L12
With sleep time: `SYNC_THREAD_SLEEP_TIME = 0.5`, if list is empty while loop in `work` finish and thread go to sleep.